### ISSUE TITLE
Folder Permission tab doesn't just redirect you to homepage when access via UploadField.

### ISSUE THREAD

**Issue by maxime-rainville** [silverstripe/silverstripe-asset-admin#761]
If you try to access the _permission_ tab for a Folder via the `UploadField`, you get redirected to your homepage.

![broken permission](https://user-images.githubusercontent.com/1168676/38341391-5a61a91a-38cc-11e8-95ff-81bc62c0ebb4.gif)

If you try managing folder permissions via `AssetsAdmin`, it works fine however.

Confirmed in 1.x-dev using Chrome 65 and Firefox 59

# Steps to reproduce
0. Access a DataObject form with with an UploadField.
0. Click the _Add from files_ link.
0. Pick a folder.
0. Click the _edit folder_ icon. This should bring up your folder details panel.
0. Click the _Permission_ tab. 

# Console logs
There's no error/warning printed to the JS console.

**Comment by maxime-rainville** [comment]
@tractorcow couldn't replicate the issue. So I double checked a bit. 

The UploadField is working properly when I add it to my Page object.

I got the issue by editing a DataObject via a ModelAdmin. I tried editing my Page object via the same Model Admin and the UploadField was still working fine on it.

So the issue appears to be specific to my DataObject. I did notice a few times that the actual permission fields were showing before I got redirected. So it looks like a simple `preventDefault` call being skipped.

```php
<?php
namespace Maxime;
use SilverStripe\ORM\DataObject;
use SilverStripe\AssetAdmin\Forms\UploadField;

class Dummy extends DataObject
{

    private static $table_name = 'dummy';

    public function getCMSFields()
    {
        $fields = parent::getCMSFields();
        $fields->addFieldToTab('Root.Main', UploadField::create('Test'));
        return $fields;
    }
}
```

**Comment by tractorcow** [comment]
Yep this bug occurs only in some browsers. It's broken in firefox but not chrome.

**Comment by tractorcow** [comment]
It's a regression in the react-bootstrap-ss upgrade. 4.0 isn't affected, but 4.2 is.

**Comment by tractorcow** [comment]
Fix at https://github.com/silverstripe/silverstripe-admin/pull/477
